subsys/shell/shell.c

```diff
@@ -1708,7 +1718,7 @@ static void help_subcmd_print(const struct shell *shell)
 			break;
 		}
 
-		u16_t len = shell_strlen(entry->syntax);
+		size_t len = shell_strlen(entry->syntax);
```

Comment:
For `shell_strlen` related changes please use casting to u16_t.

Reply:
If we are going to always cast the return of shell_strlen() why don't simply change this function to return u16_t ?

Reply:
this is very good idea!